aten/src/ATen/native/Histogram.cpp

```diff
@@ -193,7 +196,18 @@ select_outer_bin_edges(const Tensor& input, c10::optional<c10::ArrayRef<double>>
     } else if (input.numel() > 0) {
         // non-empty input
         AT_DISPATCH_FLOATING_TYPES(input.scalar_type(), "histogramdd", [&]() {
-            infer_bin_edges_from_input<scalar_t>(input, N, leftmost_edges, rightmost_edges);
+            if (input.is_mps()) {
+                // aminmax has not been implemented on mps.
+                Tensor min = at::amin(input, 0);
```

Comment:
We should register the `amin` and `amax` ops on MPS.

Reply:
Can you please register these ops first ?

Reply:
Would you allow me to add it in a follow-up PR? as this PR already involves many files, adding this op first would possibly delay the merge and result in repeated rebases, which is what I want to avoid.

Reply:
By the way we already have `amin` and `amax` ops in mps. Lacking `aminmax` for now.

Reply:
You are right, I think we are good from that standpoint.

Reply:
> Before we proceed further, imo it would be good to decide:
> 
> * Is 64-bit indexing really supported on MPS
> * How to measure perf of such kernels
> * `torch.aminmax` for MPS
> * Structured kernels to share range checks

> * `torch.aminmax` for MPS
> * Structured kernels to share range checks

@qqaatw ,  are these comments addressed ? 

> * Is 64-bit indexing really supported on MPS

@malfet this should be supported, as we have the Metal kernels. Unless there are bugs which we will fix, the 64-bit support should be there. 

> * How to measure perf of such kernels

We would like to enable `mps`  device for benchmarking code (in `torch/benchmarks` and we can use that for kernels as well.

Reply:
@kulinseth Yes, 

1. I will add `aminmax` in a follow-up PR as discussed above.
2. I've provided some [context](https://github.com/pytorch/pytorch/pull/96652#issuecomment-1489926262) about the performance stuff.
3. For structured kernels, I think it'd be great to add as a separate PR as it involves not only mps but the cpu code.